### PR TITLE
[stable/openebs]: update charts to 2.12.0 release

### PR DESCRIPTION
**Pull request by prateekpandey14**
- update chart version
- update readme
- update values
- add nfs provisioner as dependency charts

Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/openebs]`)

### CODE REVIEW

**Comment by prateekpandey14** [comment]
NFS charts dependency has been added , which needs to be merged https://github.com/openebs/dynamic-nfs-provisioner/pull/91 before this PR.

File: charts/openebs/values.yaml [hunk 1]
```diff
@@ -601,7 +601,29 @@ lvm-localpv:
...
+#  rbac:
+#    pspEnabled: false
+#
+#  nfsProvisioner:
+#    image:
+#      registry:
+#      repository: openebs/provisioner-nfs
+#      tag: 0.6.0
+#      pullPolicy: IfNotPresent
+#      enableLeaderElection: "true"
```
Comment (mynktl) [inline charts/openebs/values.yaml:626]: nit:
```suggestion
#    enableLeaderElection: "true"
```

File: charts/openebs/README.md [hunk 2]
```diff
@@ -116,26 +116,26 @@ The following table lists the configurable parameters of the OpenEBS chart and t
 | `ndmOperator.imageTag`                  | Image Tag for NDM Operator                    | `1.6.0`                                   |
 | `ndmOperator.resources`                 | Set resource limits for NDM Operator          | `{}`                                      |
 | `jiva.image`                            | Image for Jiva                                | `openebs/jiva`                            |
-| `jiva.imageTag`                         | Image Tag for Jiva                            | `2.11.0`                                  |
+| `jiva.imageTag`                         | Image Tag for Jiva                            | `2.12.0`                                  |
```
Comment (kmova) [inline charts/openebs/README.md:119]: jiva related tags needs to be updated to 2.12.1

**Comment by prateekpandey14** [comment]
>@prateekpandey14 -- looks like the changes from the following are missing:

> chore(helm-chart): update localpv-provisioner helm chart to v2.12.0 dynamic-localpv-provisioner#88

Included the localpv changes

**Review by mynktl — APPROVED** [review]

**Comment by prateekpandey14** [comment]
> @prateekpandey14 -- looks like the changes from the following are missing:

> feat(exporter): Add ndm exporters support from helm chart node-disk-manager#612

Included the NDM changes as well

File: charts/openebs/templates/_helpers.tpl [hunk 3]
```diff
@@ -41,3 +41,126 @@ Create the name of the service account to use
...
+{{- $name := default $ndmClusterExporterName .Values.ndmExporter.clusterExporter.nameOverride }}
+{{- if contains .Release.Name $name }}
+{{- $name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "openebs.exporter.name" -}}
```
Comment (kmova) [inline charts/openebs/templates/_helpers.tpl:82]: Is this duplicate of the below?

File: charts/openebs/templates/_helpers.tpl [hunk 4]
```diff
@@ -41,3 +41,126 @@ Create the name of the service account to use
...
+Define meta labels for openebs components
+*/}}
+{{- define "openebs.common.metaLabels" -}}
+chart: {{ template "openebs.chart" . }}
+heritage: {{ .Release.Service }}
+openebs.io/version: {{ .Values.release.version | quote }}
+{{- end -}}
+
+
+{{- define "openebs.cluster-exporter.name" -}}
```
Comment (kmova) [inline charts/openebs/templates/_helpers.tpl:56]: n_: change from `cluster-exporter` to `ndm-cluster-exporter`

File: charts/openebs/templates/_helpers.tpl [hunk 5]
```diff
@@ -41,3 +41,126 @@ Create the name of the service account to use
...
+{{- end }}
+{{- end }}
+
+{{- define "openebs.exporter.name" -}}
+{{- $ndmName := default .Chart.Name .Values.ndmExporter.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- $componentName := "exporter" | trunc 63 | trimSuffix "-" }}
+{{- printf "%s-%s" $ndmName $componentName | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{- define "openebs.node-exporter.name" -}}
```
Comment (kmova) [inline charts/openebs/templates/_helpers.tpl:88]: n_: change from `node-exporter` to `ndm-node-exporter`

File: charts/openebs/templates/_helpers.tpl (hunk 4 above)
Comment (prateekpandey14) [reply charts/openebs/templates/_helpers.tpl:56]: done

File: charts/openebs/templates/_helpers.tpl (hunk 3 above)
Comment (prateekpandey14) [reply charts/openebs/templates/_helpers.tpl]: fixed

**Review by kmova — APPROVED** [review]

**Pull request merged**